tests/e2e/test_spyre_cb.py

```diff
@@ -83,177 +92,389 @@ def test_cb_handling(
+    """ Scenario where it happens that some sequence gets scheduled in a way 
+    that it is misaligned with the block boundary (i.e. tkv is not a multiple 
+    of 64 at the time of prefilling). """
+
+    seqs_max_tokens = [57, 67, 9]
+    prompts_lengths = [49, 41, 47]
+    steps_add_reqs = [0, 0, 0]  # add all requests in the beginning
+
+    checked_steps = [
+        {
+            "step": 0,
+            "tkv": 0,
+            "waiting": ["0", "1", "2"],
+            "running": [],
+            "request_outputs": []
+        },
+        {
+            "step": 1,  # Prefill sequence 0
+            "tkv": 64,
+            "waiting": ["1", "2"],
+            "running": ["0"],
+            "request_outputs": ["0"]
+        },
+        {
+            "step": 2,  # Prefill sequence 1
+            "tkv": 64,  # Still 64 because this step is also a prefill
+            "waiting": ["2"],
+            "running": ["1", "0"],
+            "request_outputs": ["1"]
+        },
+        {
+            "step": 3,  # Decode sequences 0 and 1
+            "tkv": 65,
+            "waiting": ["2"],
+            "running": ["1", "0"],
+            "request_outputs": ["1", "0"]
+        },
+        {
+            # Sequence 0 finishes at step 58
+            # (start step + 2 prefills + 56 decodes - 1) = 1 + 2 + 56 - 1 = 58
+            "step": 58,
+            "tkv": 120,
+            "waiting": ["2"],
+            "running": ["1"],
+            "request_outputs": ["1", "0"],
+            "finished_requests": ["0"]
+        },
+        {
+            "step": 59,  # Prefill sequence 2
+            "tkv": 120,  # Tkv doesn't increase because it is a prefill
+            "waiting": [],
+            "running": ["2", "1"],
+            "request_outputs": ["2"]
+        },
+        {
+            "step": 60,  # Decode sequences 1 and 2
+            "tkv": 121,
+            "waiting": [],
+            "running": ["2", "1"],
+            "request_outputs": ["2", "1"]
+        },
+        {
+            # Sequence 2 finishes at step 68
+            # (start step + 1 prefill + 8 decodes - 1) = 59 + 1 + 8 - 1 = 67
+            "step": 67,
+            "tkv": 128,
+            "waiting": [],
+            "running": ["1"],
+            "request_outputs": ["2", "1"],
+            "finished_requests": ["2"]
+        },
+        {
+            "step": 68,  # Decode sequences 1
+            "tkv": 129,
+            "waiting": [],
+            "running": ["1"],
+            "request_outputs": ["1"]
+        },
+        {
+            # Sequence 1 finishes at step 69
+            # (start step + 2 prefills + 66 decodes - 1) = 2 + 2 + 66 - 1 = 69
+            "step": 69,
+            "tkv": 130,
+            "waiting": [],
+            "running": [],
+            "request_outputs": ["1"],
+            "finished_requests": ["1"]
+        },
+        {
+            # Tkv should be cleared one step later
+            "step": 70,  # Decode sequence 2
```

Comment:
nothing's decoding at this point right?
```suggestion
            "step": 70,
```